lib/command/deploy_image.rb

```diff
@@ -15,22 +16,25 @@ class DeployImage < Base
       - If the release script exits with a non-zero code, the command will stop executing and also exit with a non-zero code
     DESC
 
-    def call # rubocop:disable Metrics/MethodLength
+    def call # rubocop:disable Metrics/MethodLength, Metrics/CyclomaticComplexity
       run_release_script if config.options[:run_release_phase]
 
       deployed_endpoints = {}
 
       image = cp.latest_image
-      if cp.fetch_image_details(image).nil?
+      image_details = cp.fetch_image_details(image)
+      if image_details.nil?
         raise "Image '#{image}' does not exist in the Docker repository on Control Plane " \
               "(see https://console.cpln.io/console/org/#{config.org}/repository/#{config.app}). " \
               "Use `cpflow build-image` first."
       end
 
+      image = "#{image_details['name']}@#{image_details['digest']}" if config.options[:use_digest_ref]
```

Comment:
`image_details["name"]` contains image name and tag and we can keep the tag - it will be ignored due to existence of the digest value